core/domain/user_query_services.py

```diff
@@ -188,7 +188,7 @@ def send_email_to_qualified_users(
     email_subject: str,
     email_body: str,
     email_intent: str,
-    max_recipients: int
+    max_recipients: Optional[int]
```

Comment:
Why does this need to change?

Reply:
Because here `max_recipients` can be None (please see the schemas )
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/email_dashboard.py#L236

And inside the `send_email_to_qualified_users` method we are also accounting the None case of `max_recipients` https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/domain/user_query_services.py#L205

In my opinion, I think we are sending mails to all recipients if `max_recipients` is not provided, if it is provided then we are sending to only those last recipients.